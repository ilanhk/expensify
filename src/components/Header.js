import React from 'react';
import { NavLink } from 'react-router-dom'; 

const Header = ()=>(
    <header>
        <h1>Expensify</h1>
        <NavLink to="/" activeClassName="is-active" exact={true}>Home</NavLink>
        <NavLink to="/create" activeClassName="is-active">Add Expense</NavLink>
    </header>
);
//NavLink same like Link but has more props for navigation
// activeClassName="" is class but for NavLink

export default Header;