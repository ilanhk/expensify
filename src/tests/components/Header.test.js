import React from 'react';
import { shallow } from 'enzyme';
// import toJSON from 'enzyme-to-json'; //just using toJSON to get the correct snapshots, here are the documentation: https://github.com/adriantoine/enzyme-to-json use this once
//import ReactShallowRenderer from 'react-test-renderer/shallow'; //react-test-renderer allows us to render our components with Javascript and we can assert what got rendered
import Header from '../../components/Header';

test('should render header correctly', ()=>{
    //when using enzyme:
    const wrapper = shallow(<Header />); //shallow takes in JSX that we are trying to render
    // expect(wrapper.find('h1').text()).toBe('Expensify'); // example on what enzyme can do (can test specific elements in a component)
    // expect(toJSON(wrapper)).toMatchSnapshot(); //create a snapshot base off of the enzyme wrapper need to use toJson() once thats why this line is commented out
    expect(wrapper).toMatchSnapshot();


    //when using 'react-test-renderer/shallow' 
    // const renderer = new ReactShallowRenderer(); // need to create a new renderer first
    // renderer.render(<Header />); // to render the JSX we are trying to render
    // expect(renderer.getRenderOutput()).toMatchSnapshot(); // snapshots allow us to track changes to data over time. If you want to make a change to the component the test will fail but you will need to press 'u' to accept the changes then the test will pass.
    // console.log(renderer.getRenderOutput()); // this would return the rendered output of the JSX we put in
});

