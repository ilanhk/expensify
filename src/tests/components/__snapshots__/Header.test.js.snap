// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render header correctly 1`] = `
<header>
  <h1>
    Expensify
  </h1>
  <NavLink
    activeClassName="is-active"
    exact={true}
    to="/"
  >
    Home
  </NavLink>
  <NavLink
    activeClassName="is-active"
    to="/create"
  >
    Add Expense
  </NavLink>
</header>
`;
